sharding/src/main/scala/com/lightbend/akka/bench/sharding/scalability/ActorCountingBenchmarkMaster.scala

```diff
@@ -26,54 +26,60 @@ import akka.stream.scaladsl.{ Flow, Source }
 import com.lightbend.akka.bench.sharding.BenchSettings
 
 import scala.concurrent.duration._
+import akka.actor.Props
 
-class ActorCountingBenchmarkMaster extends Actor with ActorLogging {
+object ActorCountingBenchmarkMaster {
+  def props(sharding: ActorRef): Props =
+    Props(new ActorCountingBenchmarkMaster(sharding))
+}
+
+class ActorCountingBenchmarkMaster(sharding: ActorRef) extends Actor with ActorLogging {
   implicit val dispatcher = context.system.dispatcher
   val settings = BenchSettings(context.system)
   val addActorsBatch: Int = settings.AddActorsPerBatch
   val addActorsInterval: FiniteDuration = settings.AddActorsInterval
-  
+
   val cluster = Cluster(context.system)
-  
+
   // number of actors we sent a message to (so they should start in sharding)
   var totalStartingActorsInSharding = 0L
   // number of actors who have not yet replied back that they've started
   var pendingAliveConfirmation = 0L
   var totalAliveConfirmedActors = 0L
-  val timesWhenWeStartedBatch = new util.ArrayDeque[Long](10) 
-  
+  var batchCount = 0
+  var timesWhenWeStartedBatch = Map.empty[Int, Long]
```

Comment:
It's starting a bunch of actors in a batch and expect replies back. If such a reply is lost the ArrayDeque will be out of sync. This change adds and id (counter) to each message so that it can tracked back to which batch (starting time it was).